habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py

```diff
@@ -344,210 +343,131 @@ def render_debug_frame(
-        :param show: If True, open and display the image immediately.
+            subject_obj = get_obj_from_id(self.sim, subject)
+            if subject_obj is None:
+                raise AssertionError(
+                    f"The integer subject, '{subject}', is not a valid object_id."
+                )
+            subject = subject_obj
+        elif isinstance(subject, str):
+            if subject == "stage" or subject == "scene":
+                subject_bb = (
+                    self.sim.get_active_scene_graph()
+                    .get_root_node()
+                    .cumulative_bb
+                )
+                if cam_local_pos is None:
+                    cam_local_pos = mn.Vector3(0, 1, 0)
+            else:
+                from habitat.sims.habitat_simulator.sim_utilities import (
+                    get_obj_from_handle,
+                )
 
-        :return: a tuple containing saved filepath and the list of DebugObservations generated.
-        """
+                subject_obj = get_obj_from_handle(self.sim, subject)
+                if subject_obj is None:
+                    raise AssertionError(
+                        f"The string subject, '{subject}', is not a valid object handle or an allowed alias from ('stage', 'scene')."
+                    )
+                subject = subject_obj
+
+        if subject_bb is None:
+            # if we have gathered an object instance, process the bounding box and transform
+            if isinstance(
+                subject, habitat_sim.physics.ManagedArticulatedObject
+            ):
+                from habitat.sims.habitat_simulator.sim_utilities import (
+                    get_ao_global_bb,
+                )
 
-        # scene is best viewed from above by default
-        if cam_local_pos is None:
-            cam_local_pos = mn.Vector3(0, 1, 0)
+                obj_bb = get_ao_global_bb(subject)
+                obj_bb_local = mn.Range3D.from_center(
+                    subject.transformation.inverted().transform_point(
+                        obj_bb.center()
+                    ),
+                    obj_bb.size() / 2.0,
+                )
+                subject_bb = obj_bb_local
+                subject_transform = (
+                    subject.root_scene_node.absolute_transformation()
+                )
+            elif isinstance(subject, habitat_sim.physics.ManagedRigidObject):
+                subject_bb = subject.root_scene_node.cumulative_bb
+                subject_transform = (
+                    subject.root_scene_node.absolute_transformation()
+                )
+            else:
+                raise AssertionError(
+                    f"The subject, '{subject}', is not a supported value. Should be an object, object handle, object_id integer, or one of 'stage' or 'scene'."
```

Comment:
Would it make sense to make a `StrEnum` of subjects? That would require less searching to use the function properly as auto-completion would provide the information when needed.

Reply:
Didn't seem worth it when the only hardcoded options are "stage" and "scene". The other valid strings are object handles.

Reply:
Makes sense. I don't use these APIs often - nevermind this comment :)